en/docs/get-started/provisioning-architecture.md

```diff
@@ -77,203 +79,130 @@ mode. In the blocking mode, the authentication flow will be blocked until
+
+
+-   Extension Points - There are some extension points through which users can plug their own custom implementations with wso2 charon-core.
+    
+    -   UserManager
+    -   CharonManager
+    -   AbstractSCIMObject
+    -   AbstractAttribute
+    -   JSONEncoder
+    -   JsonDecoder
+
+    As Charon is a library, these extension points are not pluggable to the Identity Server. Instead, the clients which use Charon for the SCIM implementations can develop own custom implementations using Charon.
+
+### Charon — Implementations
+
+This contains sample implementation of the SCIM service provider to illustrate how any SCIM implementation can utilize the API and the supporting module provided by Charon.
+
+### Charon — Utils
+
+This contains the default implementations of the extension points. For example, `DefaultCharonManager` is used to initialize the extension points while `InMemoryUserManager` is the default implementation of `UserManager`. Inside the `DefaultCharonManager` the `InMemoryUserManager` instance is created and used in the charon implementations.
+
+### Charon — Samples
+This contains samples illustrating the SCIM use cases. Samples mainly contain the SCIM client side implementations which can be run against a SCIM server, and hence can also be referenced to get to know how the API provided by Charon can be used to implement SCIM client side.
+
+---
+
+## Extensible SCIM user schemas
+
+WSO2 Identity Server allows users to define their own user schema in addition to the core user schema. These configured schema are then used while creating or validating user objects. This means that custom user attributes can be passed using SCIM for identity management requirements. Follow the steps given below to add a custom attribute.
+
+For information on Extending SCIM 2.0 User Schemas, see [here](../../../extend/provisioning/extend-scim2-user-schemas/)
+
+---
+
+## cURL commands
 
 Given below are the cURL commands to add a user. The attribute name for
 the wso2Extension is **EnterpriseUser**.
 
 -   **Primary Userstore Command**
 
 ``` java
-curl -v -k --user admin:admin --data "{"schemas":[],"userName":"SureshAtt","password":"Wso2@123","EnterpriseUser":{"employeeNumber":"000111","costCenter":"111111","organization":"WSO2Org","division":"Engineering","department":"Intigration","manager":{"managerId":"111000","displayName":"Prabath"}}}" --header "Content-Type:application/json" https://localhost:9443/wso2/scim/Users
+curl -v -k --user admin:admin --data "{"schemas":[],"userName":"SureshAtt","password":"Wso2@123","EnterpriseUser":{"employeeNumber":"000111","costCenter":"111111","organization":"WSO2Org","division":"Engineering","department":"Intigration","manager":{"managerId":"111000","displayName":"Prabath"}}}" --header "Content-Type:application/json" https://localhost:9443/scim2/Users
```

Comment:
incorrect curl command. (`"EnterpriseUser"`)

Reply:
This was tested and this is the output we get.

Reply:
This is tested. The user gets created.